pkg/model/iam/tests/iam_builder_master_strict.json

```diff
@@ -0,0 +1,165 @@
+        "autoscaling:DescribeLaunchConfigurations",
+        "autoscaling:GetAsgForInstance"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sASMasterPermsAllResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "autoscaling:SetDesiredCapacity",
+        "autoscaling:TerminateInstanceInAutoScalingGroup",
+        "autoscaling:UpdateAutoScalingGroup"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Condition": {
+        "StringEquals": {
+          "ec2:ResourceTag/KubernetesCluster": "iam-builder-test.k8s.local"
+        }
+      },
+      "Sid": "kopsK8sASMasterPermsTaggedResources"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "elasticloadbalancing:AttachLoadBalancerToSubnets",
+        "elasticloadbalancing:ApplySecurityGroupsToLoadBalancer",
+        "elasticloadbalancing:CreateLoadBalancer",
+        "elasticloadbalancing:CreateLoadBalancerPolicy",
+        "elasticloadbalancing:CreateLoadBalancerListeners",
+        "elasticloadbalancing:ConfigureHealthCheck",
+        "elasticloadbalancing:DeleteLoadBalancer",
+        "elasticloadbalancing:DeleteLoadBalancerListeners",
+        "elasticloadbalancing:DescribeLoadBalancers",
+        "elasticloadbalancing:DescribeLoadBalancerAttributes",
+        "elasticloadbalancing:DetachLoadBalancerFromSubnets",
+        "elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
+        "elasticloadbalancing:ModifyLoadBalancerAttributes",
+        "elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+        "elasticloadbalancing:SetLoadBalancerPoliciesForBackendServer"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sELBMasterPermsRestrictive"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iam:ListServerCertificates",
+        "iam:GetServerCertificate"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsMasterCertIAMPerms"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetBucketLocation",
+        "s3:ListBucket"
+      ],
+      "Resource": [
+        "arn:aws:s3:::kops-tests"
+      ],
+      "Sid": "kopsK8sS3GetListBucket"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:List*"
+      ],
+      "Resource": [
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/*"
+      ],
+      "Sid": "kopsK8sS3ListBucketContents"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:Get*"
+      ],
+      "Resource": "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/*",
+      "Sid": "kopsK8sS3MasterBucketFullGet"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "kms:CreateGrant",
+        "kms:Decrypt",
+        "kms:DescribeKey",
+        "kms:Encrypt",
+        "kms:GenerateDataKey*",
+        "kms:ReEncrypt*"
+      ],
+      "Resource": [
+        "key-id-1",
```

Comment:
Can we test by adding a different key for a ebs volume?  This may be too restricting.  I am not familiar with kms perms, we need to double check if we have too many.

Reply:
@faraazkhan do you know IAM kms perms?

Reply:
As far as I understand this is a complete set to work with (and delegate use of ) a specific key.  There are some other conditions that delegate use only to aws resources (ie to EC2 so that it can mount the volumes) but I have never got good information out of AWS for exactly what is required for a minimal set here.